internal/pkg/agent/cmd/run.go

```diff
@@ -133,6 +134,22 @@ func run(override cfgOverrider, testingMode bool, fleetInitTimeout time.Duration
 	defer cancel()
 	go service.ProcessWindowsControlEvents(stopBeat)
 
+	// detect otel
+	runAsOtel, err := otel.IsOtelConfig(ctx, paths.ConfigFile())
+	if err != nil {
+		return err
+	}
+
+	if runAsOtel {
+		return otel.Run(ctx, cancel, stop, testingMode)
+	}
```

Comment:
Shouldn't we check and test the config before parsing args and in case we have to run in otel mode let the collector parse its own flags ? I had a quick look at otelcol CLI args:

```
Flags:
      --config --config=file:/path/to/first --config=file:path/to/second   Locations to the config file(s), note that only a single location can be set per flag entry e.g. --config=file:/path/to/first --config=file:path/to/second. (default [])
      --feature-gates flag                                                 Comma-delimited list of feature gate identifiers. Prefix with '-' to disable the feature. '+' or no prefix will enable the feature. (default confmap.expandEnabled,coreinternal.attraction.hash.sha256,-filter.filterlog.useOTTLBridge,-filter.filtermetric.useOTTLBridge,-filter.filterspan.useOTTLBridge,pkg.translator.prometheus.NormalizeName,-pkg.translator.prometheus.PermissiveLabelSanitization,receiver.jaeger.DisableRemoteSampling,receiver.jaegerreceiver.replaceThriftWithProto,-receiver.prometheusreceiver.UseCreatedMetric,-telemetry.disableHighCardinalityMetrics,-telemetry.useOtelForInternalMetrics,-telemetry.useOtelWithSDKConfigurationForInternalTelemetry)
  -h, --help                                                               help for otelcol
      --set func                                                           Set arbitrary component config property. The component has to be defined in the config file and the flag has a higher precedence. Array config properties are overridden and maps are joined. Example --set=processors.batch.timeout=2s
  -v, --version                                                            version for otelcol
```
I imagine that we won't support feature gates or arbitrary component property overrides in this PR... Do we have plan supporting other collector CLI args other than config ?

Reply:
it is validated in a run function of otel collector why do it twice?

Reply:
config is validated in a run function of otel collector, no need to do it twice. i don;t think we need to match collector subcommands or flags as well

Reply:
One idea might be to introduce an `otel` or `otelcol` subcommand so we can hang as many otel-specific options off it without risk of collisions with existing options.  So, for example:

```
elastic-agent otel --config
elastic-agent otel --feature-gates
...
```

or 

```
elastic-agent otelcol --config
elastic-agent otelcol --feature-gates
...
```

This way, users of the Elastic OTel Collector distribution need only change the command and the sub-command to use it over any other distribition, e.g. `otelcol ...` -> `elastic-agent otel ...` or `elastic-agent otelcol ...`.

Reply:
Pushed support for `--config` into this PR and create issues for followup work for things discussed

Reply:
@michalpristas I don't see a follow up issue for implementing the `elastic-agent otel` or `elastic-agent otelcol` subcommand as discussed above.  I checked the elastic-agent repo and our private repos as well.  

Could you please make one link to it from here?  Thanks.

Reply:
Created the issue myself: https://github.com/elastic/elastic-agent/issues/3903